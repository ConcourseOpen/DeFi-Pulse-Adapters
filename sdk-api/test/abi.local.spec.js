const { call, multiCall } = require("../dist/abi");

describe("abi", () => {
  test("call", async () => {
    const balanceOfABI = {
      constant: true,
      inputs: [
        {
          name: "_owner",
          type: "address",
        },
      ],
      name: "balanceOf",
      outputs: [
        {
          name: "balance",
          type: "uint256",
        },
      ],
      payable: false,
      stateMutability: "view",
      type: "function",
    };

    const result = await call(
      "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359",
      balanceOfABI,
      9424627,
      "0x3FfBa143f5e69Aa671C9f8e3843C88742b1FA2D9"
    );
    expect(result).toBe("3914724000000000000");
  });

  test("call with cached abi", async () => {
    const result = await call(
      "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359",
      "erc20:balanceOf",
      9424627,
      "0x3FfBa143f5e69Aa671C9f8e3843C88742b1FA2D9"
    );
    expect(result).toBe("3914724000000000000");
  });

  test("multicall with cached abi", async () => {
    const calls = [
      {
        target: "0x0000000000085d4780B73119b644AE5ecd22b376",
        params: "0x802275979B020F0ec871c5eC1db6e412b72fF20b",
      },
      {
        target: "0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359",
        params: "0xaf38668f4719ecf9452dc0300be3f6c83cbf3721",
      },
    ];
    const result = await multiCall("erc20:balanceOf", calls, 9424627);
    expect(result).toMatchSnapshot();
  });
});
