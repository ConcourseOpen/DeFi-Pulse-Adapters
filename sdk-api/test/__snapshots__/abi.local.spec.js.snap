// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abi multicall with cached abi 1`] = `
Array [
  Object {
    "input": Object {
      "params": Array [
        "0x802275979B020F0ec871c5eC1db6e412b72fF20b",
      ],
      "target": "0x0000000000085d4780B73119b644AE5ecd22b376",
    },
    "output": "9075930471597257944363",
    "success": true,
  },
  Object {
    "input": Object {
      "params": Array [
        "0xaf38668f4719ecf9452dc0300be3f6c83cbf3721",
      ],
      "target": "0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359",
    },
    "output": "14182595309792052635843",
    "success": true,
  },
]
`;
